lang/cs/Org.Apache.REEF.Bridge.Core.Tests/Fail/Task/FailTaskSuspend.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+using System;
+using System.Threading;
+using Org.Apache.REEF.Bridge.Core.Tests.Fail.Exception;
+using Org.Apache.REEF.Common.Tasks;
+using Org.Apache.REEF.Common.Tasks.Events;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Bridge.Core.Tests.Fail.Task
+{
+    internal sealed class FailTaskSuspend : ITask, IObserver<ISuspendEvent>
+    {
+        private static readonly Logger Log = Logger.GetLogger(typeof(FailTaskSuspend));
+
+        private bool _isRunning = true;
+
+        [Inject]
+        private FailTaskSuspend()
+        {
+            Log.Log(Level.Info, "task created");
+        }
+
+        public void Dispose()
+        {
+            lock (this)
+            {
+                _isRunning = false;
+                Monitor.Pulse(this);
+            }
+        }
+
+        public byte[] Call(byte[] memento)
+        {
+            lock (this)
```

Comment:
This code is repeated a number of places in the test classes -- would it help to define a base TestTask class to contain the common code?